### PR TITLE
chore: add image field to define image name when building

### PR DESCRIPTION
**Pull request by lstocchi**
This PR adds a new field `image` which defines the name to be used when building the container image.

This is related to https://github.com/projectatomic/ai-studio/issues/161 

@sallyom as we do not have much time to make additional customization let us know if those names are good for you as default values for the demo or you would prefer something else. Thank you!!

### CODE REVIEW

**Review by MichaelClifford — APPROVED** [review]

**Review by jeffmaury — APPROVED** [review]
LGTM

File: chatbot-langchain/ai-studio.yaml [hunk 1]
```diff
@@ -15,11 +15,13 @@ application:
         - amd64
       ports:
         - 8001
+      image: quay.io/redhat-et/llamacpp-server
```
Comment (sallyom) [inline chatbot-langchain/ai-studio.yaml]: Currently, the model-server built w/ gh actions is [redhat-etlocallm-model-service](https://quay.io/repository/redhat-et/locallm-model-service) - so should be `quay.io/redhat-et/locallm-model-service:latest`

File: chatbot-langchain/ai-studio.yaml [hunk 2]
```diff
@@ -15,11 +15,13 @@ application:
...
       contextdir: .
       containerfile: builds/Containerfile
       arch:
         - arm64
         - amd64
       ports:
-        - 8501
\ No newline at end of file
+        - 8501
+      image: quay.io/redhat-et/streamlit-chat-app
```
Comment (sallyom) [inline chatbot-langchain/ai-studio.yaml]: This one should be `quay.io/redhat-et/locallm-chatbot:latest` (we can rename, but that's what it is atm)

File: code-generation/ai-studio.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@ application:
         - amd64
       ports:
         - 8001
+      image: quay.io/redhat-et/llamacpp-server
```
Comment (sallyom) [inline code-generation/ai-studio.yaml]: `quay.io/redhat-et/locallm-model-service:latest`

File: code-generation/ai-studio.yaml [hunk 4]
```diff
@@ -23,3 +24,4 @@ application:
         - amd64
       ports:
         - 8501
+      image: quay.io/redhat-et/codegen-app
```
Comment (sallyom) [inline code-generation/ai-studio.yaml]: `quay.io/redhat-et/locallm-codegen:latest`

File: object_detection/ai-studio.yaml [hunk 5]
```diff
@@ -15,11 +15,13 @@ application:
         - amd64
       ports:
         - 8000
+      image: quay.io/redhat-et/object-detection-server
```
Comment (sallyom) [inline object_detection/ai-studio.yaml:18]: this one we haven't added to actions yet! (I'll quickly do that now)

File: object_detection/ai-studio.yaml [hunk 6]
```diff
@@ -15,11 +15,13 @@ application:
...
       contextdir: ./client
       containerfile: Containerfile
       arch:
         - arm64
         - amd64
       ports:
-        - 8501
\ No newline at end of file
+        - 8501
+      image: quay.io/redhat-et/object-detection-client
```
Comment (sallyom) [inline object_detection/ai-studio.yaml:27]: not set up in actions yet, we need to add that!

File: rag-langchain/ai-studio.yaml [hunk 7]
```diff
@@ -15,6 +15,7 @@ application:
         - amd64
       ports:
         - 8001
+      image: quay.io/redhat-et/llamacpp-server
```
Comment (sallyom) [inline rag-langchain/ai-studio.yaml]: quay.io/redhat-et/locallm-model-service:latest

File: rag-langchain/ai-studio.yaml [hunk 8]
```diff
@@ -24,11 +25,13 @@ application:
         - amd64
       ports:
         - 8000
+      image: quay.io/redhat-et/chromadb-server
```
Comment (sallyom) [inline rag-langchain/ai-studio.yaml]: quay.io/redhat-et/locallm-chromadb:latest

File: rag-langchain/ai-studio.yaml [hunk 9]
```diff
@@ -24,11 +25,13 @@ application:
...
       contextdir: .
       containerfile: builds/Containerfile
       arch:
         - arm64
         - amd64
       ports:
-        - 8501
\ No newline at end of file
+        - 8501
+      image: quay.io/redhat-et/rag-inference-app
```
Comment (sallyom) [inline rag-langchain/ai-studio.yaml]: quay.io/redhat-et/locallm-rag:latest

File: summarizer-langchain/ai-studio.yaml [hunk 10]
```diff
@@ -15,11 +15,13 @@ application:
         - amd64
       ports:
         - 8001
+      image: quay.io/redhat-et/llamacpp-server
```
Comment (sallyom) [inline summarizer-langchain/ai-studio.yaml]: quay.io/redhat-et/locallm-model-service:latest

File: summarizer-langchain/ai-studio.yaml [hunk 11]
```diff
@@ -15,11 +15,13 @@ application:
...
       contextdir: .
       containerfile: builds/Containerfile
       arch:
         - arm64
         - amd64
       ports:
-        - 8501
\ No newline at end of file
+        - 8501
+      image: quay.io/redhat-et/streamlit-summary-app
```
Comment (sallyom) [inline summarizer-langchain/ai-studio.yaml]: quay.io/redhat-et/locallm-text-summarizer:latest

File: chatbot-langchain/ai-studio.yaml (hunk 1 above)
Comment (lstocchi) [reply chatbot-langchain/ai-studio.yaml]: my bad, i didn't check the gh actions. Fixed 👍 Thanks!!

File: chatbot-langchain/ai-studio.yaml (hunk 2 above)
Comment (lstocchi) [reply chatbot-langchain/ai-studio.yaml]: It's the same for us, so what's good for you, it's good for us 👍

**Comment by lstocchi** [comment]
@sallyom updated everything. Let me know when you have the name for the object_detection images. Thanks!

File: object_detection/ai-studio.yaml (hunk 6 above)
Comment (sallyom) [reply object_detection/ai-studio.yaml:27]: `quay.io/redhat-et/locallm-object-detection-client:latest`

File: object_detection/ai-studio.yaml (hunk 5 above)
Comment (sallyom) [reply object_detection/ai-studio.yaml:18]: quay.io/redhat-et/locallm-object-detection-server:latest

**Comment by sallyom** [comment]
LGTM - I'll upate the object-detection image names in a follow-up when I add the actions for those, let's merge this in now!

**Comment by jeffmaury** [comment]
As a general pattern, as we will include the recipes using a reference or a commit hash, we should not use latest for OCI image tags

**Pull request merged**